docs/usage/use-specification-repository-pattern.md

```diff
@@ -7,6 +7,61 @@ nav_order: 2
 
 # How to use Specifications with the Repository Pattern
 
-Specifications shine when combined with the [Repository Pattern](https://deviq.com/design-patterns/repository-pattern), a sample generic implementation of which is included in this NuGet package.
+Specifications shine when combined with the [Repository Pattern](https://deviq.com/design-patterns/repository-pattern), a sample generic implementation of which is included in this NuGet package. For the purpose of this walkthrough, the repository can be thought of as a simple data access abstraction over a collection of entities. In this example, the entity for a `Hero` and it's repository interface are defined below.
 
-(todo)
+```csharp
+public class Hero
+{
+    public string Name { get; set; }
+    public string SuperPower { get; set; }
+    public bool IsAlive { get; set; }
+    public bool IsAvenger { get; set; }
+}
+
+public interface IHeroRepository
+{
+    List<Hero> GetAllHeroes();
+}
+
+```
+
+A Specification can be defined for the `Hero` entity when filtering is required for the collection of all heroes. Note any fields that are needed to filter the heroes are passed to the Specification's constructor where the query logic should be implemented. In this case, a Specification is defined that filters heroes by whether the hero is alive and is an Avenger.
+
+```csharp
+public class HeroByIsAliveAndIsAvengerSpec : Specification<Hero>
+{
+    public HeroByIsAliveAndIsAvengerSpec(bool isAlive, bool isAvenger)
+    {
+        Query.Where(h => h.IsAlive == isAlive && h.IsAvenger == isAvenger);
+    }
+}
+```
+
+With the Specification and Repository defined, it is now possible to define a `GetHeroes` method that can take a `HeroRepository` as a parameter along with the filtering conditions and produce a filtered collection of heroes. Applying the Repository to the Specification is done using the `Evaluate` method on the Specification class which takes a `IEnumerable<T>` as a parameter. This should mirror the kind of methods typically found on Controllers or [Api Endpoints](https://github.com/ardalis/ApiEndpoints) where the IHeroRepository might be supplied via Dependency Injection to the class's constructor rather than passed as a parameter.
+
+```csharp
+public IEnumerable<Hero> GetHeroes(IHeroRepository repository, bool isAlive, bool isAvenger)
+{
+    var specification = new HeroByIsAliveAndIsAvengerSpec(isAlive, isAvenger);
+
+    return specification.Evaluate(repository.GetAllHeroes());
```

Comment:
We'd only want to use `Evaluate` on existing in-memory collections. I don't think we want to propose that one use it on a repository's results, as that would always be performed in-memory and would be less efficient than filtering the data within the data source itself.

We do need docs and samples of `Evaluate`, but this document should show how one can pass a `Specification` to a repository method and have the repository use that specification to perform querying logic internally. Essentially the code here should be similar to but separate from the built-in repository, for users who want to add support for specifications to their own repository rather than inheriting from the one we provide in this package.

Reply:
Updated to show how a basic example of how to modify an existing repository to apply specifications.